File: 스택/큐/기능개발/기능개발.js
function solution(progresses, speeds) {
    const result = []
    let curProgresses = [...progresses]
    while (true) {
        curProgresses = curProgresses.map((e, i) => e + speeds[i])
        let complete = 0
        for (const [i, e] of curProgresses.entries()) {
            if (e >= 100) complete++
            else break
        }
        if (complete > 0) {
            curProgresses.splice(0, complete)
            speeds.splice(0, complete)
            result.push(complete)
        }
        if (curProgresses.length <= 0) return result
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
import java.util.stream.*;

class Solution {
	public int[] solution(int[] progresses, int[] speeds) {
		ArrayList<Integer> list = Arrays
			.stream(progresses)
			.boxed()
			.collect(Collectors.toCollection(ArrayList::new));

		List<Integer> result = new ArrayList<>();
		int pointer = 0;
		while (pointer < list.size()) {
			int cnt = 0;
			while (pointer < list.size() && list.get(pointer) >= 100) {
				pointer++;
				cnt++;
			}
            if(cnt > 0) {
                result.add(cnt);
            }
			for (int i = pointer; i < speeds.length; i++) {
				list.set(i, list.get(i) + speeds[i]);
			}
		}
		return result.stream().mapToInt(Integer::intValue).toArray();
	}
}
